tests/D2L.CodeStyle.Analyzers.Test/Common/MutabilityInspectorTests.cs

```diff
@@ -67,120 +67,190 @@ public class MutabilityInspectorTests {
+				null,
+				$"{RootNamespace}.foo",
+				MutabilityTarget.Type,
+				MutabilityCause.IsNotSealed
+			);
+
+			var actual = m_inspector.InspectType( type );
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( type ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_SealedClass_False() {
+		public void InspectType_SealedClass_False() {
 			var type = Type( "sealed class foo {}" );
+			var expected = MutabilityInspectionResult.NotMutable();
 
-			Assert.IsFalse( m_inspector.IsTypeMutable( type ) );
+			var actual = m_inspector.InspectType( type );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_LooksAtFieldsInType() {
+		public void InspectType_LooksAtMembersInDeclaredType() {
 			var field = Field( "public string random" );
 			Assert.IsTrue( m_inspector.IsFieldMutable( field ) );
+			var expected = MutabilityInspectionResult.Mutable(
+				"random",
+				"System.String",
+				MutabilityTarget.Member,
+				MutabilityCause.IsNotReadonly
+			);
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( field.ContainingType ) );
+			var actual = m_inspector.InspectType( field.ContainingType );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_LooksAtMutabilityOfTypeOfField() {
+		public void InspectType_LooksAtMembersInExternalType() {
 			var field = Field( "public readonly System.Text.StringBuilder random" );
 			Assert.IsFalse( m_inspector.IsFieldMutable( field ) );
+			var expected = MutabilityInspectionResult.Mutable(
+				"Capacity",
+				"System.Int32",
+				MutabilityTarget.Member,
+				MutabilityCause.IsNotReadonly
+			);
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( field.ContainingType ) );
-		}
+			var actual = m_inspector.InspectType( field.Type );
 
-		[Test]
-		public void IsTypeMutable_LooksAtPropertiesInType() {
-			var prop = Property( "public string random { get; set; }" );
-			Assert.IsTrue( m_inspector.IsPropertyMutable( prop ) );
-
-			Assert.IsTrue( m_inspector.IsTypeMutable( prop.ContainingType ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_LooksAtMutabilityOfTypeOfProperty() {
-			var property = Property( "public System.Text.StringBuilder random { get; }" );
-			Assert.IsFalse( m_inspector.IsPropertyMutable( property ) );
+		public void InspectType_LooksFieldsInType() {
+			var field = Field( "public readonly System.Text.StringBuilder random" );
+			Assert.IsFalse( m_inspector.IsFieldMutable( field ) );
+			var expected = MutabilityInspectionResult.Mutable(
+				"random.Capacity",
+				"System.Int32",
+				MutabilityTarget.Member,
+				MutabilityCause.IsNotReadonly
+			);
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( property.Type ) );
+			var actual = m_inspector.InspectType( field.ContainingType );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_NonExistentType_ThrowsException() {

```

Comment:
Killed this test as it was implemented incorrectly. I will add it back and implement it correctly, as we don't currently test that anywhere (I thought we did in `UnsafeStaticsAnalyzerTests`).